dashboard/app/models/user.rb

```diff
@@ -462,16 +459,16 @@ def email_matches_for_oauth_upgrade
   validates_presence_of :email_preference_form_kind, if: -> {email_preference_opt_in.present?}
 
   # Validations for adding parent email notifications
-  before_validation :parent_email_preference_setup, if: :parent_email_preference_opt_in_required?
-  validates_inclusion_of :parent_email_preference_opt_in, in: %w(yes no), if: :parent_email_preference_opt_in_required?
-  validates_presence_of :parent_email_preference_email, if: :parent_email_preference_opt_in_required?
-  validates_presence_of :parent_email_preference_request_ip, if: :parent_email_preference_opt_in_required?
-  validates_presence_of :parent_email_preference_source, if: :parent_email_preference_opt_in_required?
+  before_validation :parent_email_preference_setup, if: -> {parent_email_preference_email.present? && user_type == 'student'}
+  validates_inclusion_of :parent_email_preference_opt_in, in: %w(yes no), if: :update_parent_email_preference_opt_in?
+  validates_presence_of :parent_email_preference_email, if: :update_parent_email_preference_opt_in?
+  validates_presence_of :parent_email_preference_request_ip, if: :update_parent_email_preference_opt_in?
+  validates_presence_of :parent_email_preference_source, if: :update_parent_email_preference_opt_in?
 
-  def parent_email_preference_opt_in_required?
+  def update_parent_email_preference_opt_in?
     # parent_email_preference_opt_in_required is a checkbox which either has the value '0' or '1'
     # user_type 'student' is the only type which supports have a parent_email associated with it.
-    parent_email_preference_opt_in_required == '1' && user_type == 'student'
+    (parent_email_preference_opt_in.present? || parent_email_preference_opt_in_required == '1') && user_type == 'student'
```

Comment:
I don't think this is right because `parent_email_preference_opt_in.present?` does not indicate whether or not to the sign up flow will create an EmailPreference.
Imagine these steps during the sign_up
1. Student is on the final sign up page
2. Student clicks the check box "I'm a parent...."
3. Student fills out a parent email and clicks "Yes" on the Yes/No radio buttons.
4. Student **unchecks** the "I'm a parent...".
5. Student  continues filling out Display Name, etc and clicks Submit

I might be mistaken, but I believe that this will still result in the parent email notification being created because `display: hidden` fields will still be submitted. That was my original reasoning for depending on checking the value of the checkbox before doing anything parent email preference related.

This edge case should have been covered by the unit test: `creating a student with no parent email preference should not create an email preference for the parent`
However the unit test doesn't seem to be failing in this PR, so I must have written it incorrectly or I'm reading the code wrong.

Reply:
Please let me know what you think. I will try to figure this out myself too.

Reply:
Yeah I figured out I wrote the unit tests wrong. I have updated them in staging. You can pull the changes now. This tests will probably start failing though. We will need to talk about this use case and how to fix it.
https://github.com/code-dot-org/code-dot-org/pull/34110

Reply:
Yeah this part was the biggest challenge in this PR because it feels like this path is fundamentally changed a bit. I'll merge your change and see where we are

Reply:
The only test that fails now is `test_creating_a_student_with_no_parent_email_preference_should_not_create_an_email_preference_for_the_parent`. From the perspective of this PR, that test isn't correct -- it's possible for opt in not to be required. But from the perspective of account creation it's still necessary...

I'm going to try what you suggested below.